Core/src/TrackFitting/MbfSmoother.cpp

```diff
@@ -11,68 +11,72 @@
 namespace Acts {
 
 void MbfSmoother::calculateSmoothed(InternalTrackState& ts,
-                                    const BoundMatrix& big_lambda_hat,
-                                    const BoundVector& small_lambda_hat) const {
+                                    const BoundMatrix& bigLambdaHat,
+                                    const BoundVector& smallLambdaHat) const {
   ts.smoothedCovariance = ts.filteredCovariance - ts.filteredCovariance *
-                                                      big_lambda_hat *
+                                                      bigLambdaHat *
                                                       ts.filteredCovariance;
-  ts.smoothed = ts.filtered - ts.filteredCovariance * small_lambda_hat;
+  ts.smoothed = ts.filtered - ts.filteredCovariance * smallLambdaHat;
 }
 
 void MbfSmoother::visitNonMeasurement(const InternalTrackState& ts,
-                                      BoundMatrix& big_lambda_hat,
-                                      BoundVector& small_lambda_hat) const {
-  const auto F = ts.jacobian;
+                                      BoundMatrix& bigLambdaHat,
+                                      BoundVector& smallLambdaHat) const {
+  const Acts::BoundMatrix& F = ts.jacobian;
 
-  big_lambda_hat = F.transpose() * big_lambda_hat * F;
-  small_lambda_hat = F.transpose() * small_lambda_hat;
+  bigLambdaHat = F.transpose() * bigLambdaHat * F;
+  smallLambdaHat = F.transpose() * smallLambdaHat;
 }
 
 void MbfSmoother::visitMeasurement(const InternalTrackState& ts,
-                                   BoundMatrix& big_lambda_hat,
-                                   BoundVector& small_lambda_hat) const {
+                                   BoundMatrix& bigLambdaHat,
+                                   BoundVector& smallLambdaHat) const {
   assert(ts.measurement.has_value());
 
-  const auto& measurement = ts.measurement.value();
+  const InternalTrackState::Measurement& measurement = ts.measurement.value();
+  const Acts::BoundMatrix& F = ts.jacobian;
```

Comment:
after the discussion we had with stephen I remembered that we use map types sot this might have a performance impact

Reply:
so you are suggesting what exactly? reference or non reference?

Reply:
I think you first suggestion was correct - sorry for the back and forth. `InternalTrackState::Jacobian` - personally I would bind to it by value but I don't think it makes a difference